File: 백준/Bronze/27161. 크레이지 타임/크레이지 타임.py
n = int(input())
x = 1; h = 0
for _ in range(n):
    c, i = input().split()
    h = ((h - 1) + x) % 12 + 1
    ch = h == int(i)
    rh = c == 'HOURGLASS'
    sy = 'NO'
    if ch != rh:
        if ch: sy = 'YES'
        else: x *= -1
    print(h, sy)

File: 백준/Bronze/27161. 크레이지 타임/크레이지 타임.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int x = 1;
        int h = 0;
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine()," ");
            String c = st.nextToken();
            int k = Integer.parseInt(st.nextToken());
            h = h + x;
            if (h > 12) { h -= 12;
            } else if (h < 1) { h += 12;
            }
            boolean ch = h == k;
            boolean rh = Objects.equals(c, "HOURGLASS");
            String sy = "NO";
            if (ch != rh) {
                if (ch) { sy = "YES";
                } else { x *= -1; }
            }
            System.out.println(h + " " + sy);
        }
    }
}